apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs

```diff
@@ -389,9 +389,7 @@ private object ConvertToClrTypeImpl(Value wireValue, System.Type targetClrType,
             {
                 if (targetClrType == typeof(IDictionary))
                 {
-#pragma warning disable DE0006
-                    targetClrType = typeof(Hashtable);
-#pragma warning restore DE0006
+                    targetClrType = typeof(Dictionary<string, object>);
```

Comment:
I don't know this code well...
But what if `targetClrType` is `IDictionary<string, object>`, the `Activator.Createinstance` call below will fail.
Or is `IDictionary<,>` unsupported here?

Reply:
It looks like `IDictionary<,>` has never been supported as a target type - ditto `IList<T>`. We can add support for that later if it would be desirable (it sounds useful) without it being a breaking change.